datalad/support/sshconnector.py

```diff
@@ -450,7 +314,351 @@ def get_git_version(self):
 
 
 @auto_repr
-class SSHManager(object):
+class SingleProcessSSHConnection(BaseSSHConnection):
+    """Representation of an SSH connection.
+
+    The connection is opened for execution of a single process, and closed
+    as soon as the process end.
+    """
+    def __init__(self, sshri, **kwargs):
+        """Create a connection handler
+
+        The actual opening of the connection is performed on-demand.
+
+        Parameters
+        ----------
+        sshri: SSHRI
+          SSH resource identifier (contains all connection-relevant info),
+          or another resource identifier that can be converted into an SSHRI.
+        **kwargs
+          Pass on to BaseSSHConnection
+        """
+        super().__init__(sshri, **kwargs)
+        self._ssh_open_args += [
+            # we presently do not support any interactive authentication
+            # at the time of process execution
+            '-o', 'PasswordAuthentication=no',
+            '-o', 'KbdInteractiveAuthentication=no',
+        ]
+
+    def __call__(self, cmd, options=None, stdin=None, log_output=True):
+        """Executes a command on the remote.
+
+        It is the callers responsibility to properly quote commands
+        for remote execution (e.g. filename with spaces of other special
+        characters). Use the `sh_quote()` from the module for this purpose.
+
+        Parameters
+        ----------
+        cmd: str
+          command to run on the remote
+        options : list of str, optional
+          Additional options to pass to the `-o` flag of `ssh`. Note: Many
+          (probably most) of the available configuration options should not be
+          set here because they can critically change the properties of the
+          connection. This exists to allow options like SendEnv to be set.
+
+        Returns
+        -------
+        tuple of str
+          stdout, stderr of the command run.
+        """
+        # there is no dedicated "open" step, put all args together
+        ssh_cmd = ["ssh"] + self._ssh_open_args + self._ssh_args
+        return self._exec_ssh(
+            ssh_cmd,
+            cmd,
+            options=options,
+            stdin=stdin,
+            log_output=log_output)
+
+    def is_open(self):
+        return False
+
+    def open(self):
+        return False
```

Comment:
no docstring here, only in the "original" implementation where it says

```
        bool
          True when SSH reports success opening the connection, False when
          a ControlMaster for an open connection already exists.
```
so -- does it mean that we the ControlMaster always exists there?  should docs be tuned up or should we introduce returning `None` if there is no `ControlMaster`?